src/components/page-layout/Layout.js

```diff
@@ -32,49 +30,61 @@ const createInitFoodList = () => {
 };
 
 const Layout = ({ children }) => {
-  const data = useStaticQuery(graphql`
-    query SiteTitleQuery {
-      site {
-        siteMetadata {
-          title
-        }
-      }
-    }
-  `);
-
   const [cartState, setCartState] = useState(createInitFoodList());
   const [modalOpen, setModalOpen] = useState(false);
+  const [headerTop, setHeaderTop] = useState('-150px');
 
   const toggleModal = () => {
     setModalOpen(!modalOpen);
   };
 
+  const toggleHeaderTop = () => {
+    if (headerTop === '0' || headerTop === 0) {
+      setHeaderTop('-150px');
+    } else {
+      setHeaderTop('0');
+    }
+  };
+
+  window.addEventListener('scroll', () => {
+    const st = window.pageYOffset || document.documentElement.scrollTop;
+    if (st > 400 && headerTop !== '0') {
+      setHeaderTop('0');
+    }
+  });
+
   return (
     <ThemeProvider theme={theme}>
-      <Header siteTitle={data.site.siteMetadata.title} />
-      <NavigationBar toggleModal={toggleModal} />
-      <CartModal cartState={cartState} open={modalOpen} toggleModal={toggleModal} />
-      <div className={styles.body}>
-      <Grid
-        container
-        justify="center"
-        alignItems="center"
-        sm={12}
-        className={styles.layoutBody}
-      >
-          {React.cloneElement(children, { cartState, setCartState })}
-      </Grid>
-        <footer
-          className={styles.layoutFooter}
+      <div className={styles.fullPage}>
+        <div
+          className={styles.layoutHeaderOn}
+          style={{ top: headerTop }}
+        >
+          <NavigationBar resetHeaderTop={toggleHeaderTop} toggleModal={toggleModal} />
+          <CartModal cartState={cartState} open={modalOpen} toggleModal={toggleModal} />
+        </div>
+        <div className={styles.body}>
+        <Grid
+          container
+          justify="center"
+          alignItems="center"
+          sm={12}
+          className={styles.layoutBody}
         >
-          <p>
-            ©2020, A <a href="https://www.viget.com/">Viget</a> Project
-          </p>
-          <p>
-            Liam Becker, Jennifer Montoya, William Dinneen, Jackson Doyle, Mika
-            Byar
-          </p>
-        </footer>
+            {React.cloneElement(children, { cartState, setCartState, toggleHeaderTop })}
+        </Grid>
+          <footer
+            className={styles.layoutFooter}
+          >
+            <p>
+              ©2020, A <a href="https://www.viget.com/">Viget</a> Project
```

Comment:
Doug left a comment about this that we haven't addressed yet, just wanna make sure it's still on our radar

https://github.com/vigetlabs/EcoCart/pull/16#discussion_r452530486